omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierSourceResource.java

```diff
@@ -0,0 +1,396 @@
+            Object batchSize = postBody.get("batchSize");
+            Object minPoolSize = postBody.get("minPoolSize");
+            Object sourceUuid = postBody.get("sourceUuid");
+            Object sequential = postBody.get("sequential");
+            Object refillWithScheduledTask = postBody.get("refillWithScheduledTask");
+
+            if (sourceUuid == null || StringUtils.isBlank(sourceUuid.toString())) {
+                errors.add("identifier souce");
+            }
+            else{
+                IdentifierSource poolIdentifierSource = Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(sourceUuid.toString());
+                if(poolIdentifierSource != null){
+                    identifierSource.setSource(poolIdentifierSource);
+                }else{
+                    errors.add("identifier souce");
+                }
+            }
+            if (description != null && StringUtils.isNotBlank(description.toString())) {
+                identifierSource.setDescription(description.toString());
+            }
+            if (batchSize != null && StringUtils.isNotBlank(batchSize.toString())) {
+                if(StringUtils.isNumeric(batchSize.toString())) {
+                    identifierSource.setBatchSize(Integer.parseInt(batchSize.toString()));
+                }else {
+                    errors.add("batch size");
+                }
+            }
+            if (minPoolSize != null && StringUtils.isNotBlank(minPoolSize.toString())) {
+                if(StringUtils.isNumeric(minPoolSize.toString())) {
+                    identifierSource.setMinPoolSize(Integer.parseInt(minPoolSize.toString()));
+                }else {
+                    errors.add("min pool size");
+                }
+            }
+            if (sequential != null && StringUtils.isNotBlank(sequential.toString())) {
+                if(parseBoolean(sequential.toString()) != null) {
+                    identifierSource.setSequential(parseBoolean(sequential.toString()));
+                }else {
+                    errors.add("sequential");
+                }
+            }
+            if (refillWithScheduledTask != null && StringUtils.isNotBlank(refillWithScheduledTask.toString())) {
+                if(parseBoolean(refillWithScheduledTask.toString()) != null) {
+                    identifierSource.setSequential(parseBoolean(refillWithScheduledTask.toString()));
+                }else {
+                    errors.add("refill");
+                }
+            }
+            if(errors.toString().length() > 2) {
+                throw new ValidationException("The values of the following inputs are missing or invalid: " + errors.toString());
+            }
+
+            identifierSource.setIdentifierType(patientIdentifierType);
+            identifierSource.setName(name.toString());
+            if(identifierSource.getBatchSize() > 0 && identifierSource.getSource() != null){
```

Comment:
Do you need to do this? Shouldn't you just save.

Reply:
Removed